large-message-core/src/main/java/com/bakdata/kafka/CompressionType.java

```diff
@@ -0,0 +1,142 @@
+package com.bakdata.kafka;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+
+import org.apache.kafka.common.errors.SerializationException;
+import org.apache.kafka.common.record.RecordBatch;
+import org.apache.kafka.common.utils.BufferSupplier;
+import org.apache.kafka.common.utils.ByteBufferOutputStream;
+
+import software.amazon.awssdk.utils.IoUtils;
+
+public enum CompressionType {
+    NONE(0, "none") {
+        @Override
+        public byte[] compress(byte[] bytes) {
+            return bytes;
+        }
+
+        @Override
+        public byte[] decompress(byte[] bytes) {
+            return bytes;
+        }
+    },
+
+    GZIP(1, "gzip") {
+        @Override
+        public byte[] compress(byte[] bytes) {
+            return CompressionType.compress(org.apache.kafka.common.record.CompressionType.GZIP, bytes);
+        }
+
+        @Override
+        public byte[] decompress(byte[] bytes) {
+            return CompressionType.decompress(org.apache.kafka.common.record.CompressionType.GZIP, bytes);
+        }
+    },
+
+    SNAPPY(2, "snappy") {
+        @Override
+        public byte[] compress(byte[] bytes) {
+            return CompressionType.compress(org.apache.kafka.common.record.CompressionType.SNAPPY, bytes);
+        }
+
+        @Override
+        public byte[] decompress(byte[] bytes) {
+            return CompressionType.decompress(org.apache.kafka.common.record.CompressionType.SNAPPY, bytes);
+        }
+    },
+
+    LZ4(3, "lz4") {
+        @Override
+        public byte[] compress(byte[] bytes) {
+            return CompressionType.compress(org.apache.kafka.common.record.CompressionType.LZ4, bytes);
+        }
+
+        @Override
+        public byte[] decompress(byte[] bytes) {
+            return CompressionType.decompress(org.apache.kafka.common.record.CompressionType.LZ4, bytes);
+        }
+    },
+
+    ZSTD(4, "zstd") {
+        @Override
+        public byte[] compress(byte[] bytes) {
+            return CompressionType.compress(org.apache.kafka.common.record.CompressionType.ZSTD, bytes);
+        }
+
+        @Override
+        public byte[] decompress(byte[] bytes) {
+            return CompressionType.decompress(org.apache.kafka.common.record.CompressionType.ZSTD, bytes);
+        }
+    };
+
+    private static byte[] compress(org.apache.kafka.common.record.CompressionType compressionType, byte[] bytes) {
+        ByteBufferOutputStream outStream = new ByteBufferOutputStream(bytes.length);
```

Comment:
please mark as final

Reply:
There is also a code style linked in the README

Reply:
I tried this:
```
~/projects/kafka-large-message-serde$ bash -c "$(curl -sL https://raw.githubusercontent.com/bakdata/bakdata-code-styles/master/install.sh)"
Downloading code-styles...

Could not find IntelliJ setting folder and skipping IntelliJ project setup.
If you use IntelliJ, manually setup by going to preferences, and apply '' in codestyle and '' in inspection.
```
but could not immediately get this to work on my machine (ubuntu 22.04).
Are there any other code style violations?

Reply:
Hi @domsj, we pushed a fix so that issue should hopefully be resolved